File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        
        if (nums.empty())
          return 0;

        vector<int> dp(nums.size(), 1);

        for (int i = 1; i < nums.size(); ++i) {
          for (int j = 0; j < i; ++j) {
            if (nums[j] < nums[i])
              dp[i] = max(dp[i], dp[j] + 1);
            }
        }


        int result = INT_MIN;
        
        for (int num : dp) {
            result = max(num, result);    
        }
        
        return result;
    }
    
};

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        
        if not nums:
            return 0
        
        dp = [1] * len(nums)
        
        for i in range(1, len(nums)):
            for j in range(i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j] + 1)
        
        result = 0
        
        for num in dp:
            result = max(result, num)
        
        return result
        